Refactor export structure

guilpier-code opened the pull request:
This work is in the scope of issue #979 and is a preliminary refactoring work. 

The names of export files **interconnexion.txt**, **Area.txt** and **Variables.txt** don't need to wear a simulation period (which simulation week, year or even day). Thus, these names does not need to be built with the function **getFilenameWithExtension** that automatically add a simulation period hint in the name of an export file.
These files names composition are now very simple.

guilpier-code committed:
Correct enum items for MPS

guilpier-code committed:
Refactor interconnexion export

guilpier-code committed:
Refactor areas export

guilpier-code committed:
Refactor areas export - step 2

guilpier-code committed:
Refactor areas export - step 3

guilpier-code committed:
opt export structure : export file name simplified

guilpier-code commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp:
```diff
@@ -1238,13 +1234,11 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
     {
-        OPT_ExportVariables(*study, varname, "variables", "txt", numSpace);
-
-        // TODO : for now empty constraints.txt file needed
```
Export file **constraints.txt** is empty and useless, so I removed it.
If it turns out to be useful at some point, we will add it easily.

guilpier-code commented on src/solver/optimisation/opt_export_structure.cpp:
```diff
@@ -65,43 +62,31 @@ const std::initializer_list<std::string>& getNames<ExportStructDict>()
 } // namespace Data
 } // namespace Antares
 
-void OPT_ExportInterco(const Antares::Data::Study& study,
```
Here we used to pass the whole study. This is unclear as we only need its writer.

flomnes commented on src/solver/optimisation/opt_export_structure.cpp:
```diff
@@ -140,18 +125,14 @@ void OPT_Export_add_variable(std::vector<std::string>& varname,
 {
     Yuni::Clob Flot;
-    auto filename = getFilenameWithExtension(fileName, fileExtension, numSpace);
+    std::string filename = "Variables.txt";
```
```suggestion
    std::string filename = "variables.txt";
```

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp:
```diff
@@ -1238,13 +1234,11 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
-        std::vector<std::string> conname;
-        OPT_ExportVariables(*study, conname, "constraints", "txt", numSpace);
+        OPT_ExportInterco(study->resultWriter, ProblemeHebdo);
+        OPT_ExportAreaName(study->areas, study->resultWriter);
+        OPT_ExportVariables(study->resultWriter, varname);
```
[Style] 
Writer = 1st argument, content = 2nd argument

flomnes commented on src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp:
```diff
@@ -92,12 +92,8 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
-    {
-        OPT_ExportInterco(*study, ProblemeHebdo, numSpace);
-        OPT_ExportAreaName(*study, ProblemeHebdo, numSpace);
-    }
+    bool firstWeekOfSimulation = ProblemeHebdo->firstWeekOfSimulation;
+    bool doWeExport = exportStructure && firstWeekOfSimulation;
```
All these `bool` are `const`

flomnes commented on src/libs/antares/study/fwd.cpp:
```diff
@@ -347,9 +347,9 @@ std::string mpsExportStatusToString(const mpsExportStatus& mps_export_status)
         return "none";
-    case mpsExportStatus::EXPORT_FIRST_OPIM:
+    case mpsExportStatus::EXPORT_FIRST_OPTIM:
         return "optim-1";
-    case mpsExportStatus::EXPORT_SECOND_OPIM:
+    case mpsExportStatus::EXPORT_SECOND_OPTIM:
```
Nobody ever noticed that typo, good job !

flomnes reviewed (changes requested):
LGTM

- Minor changes
- Files variable-x-y.txt are not identical

```
omnesflo@gm0winl370:~/short/029 Pumped storage plant - 2/output/20221220-1610eco$ md5sum variables-* | sort
3477bd34fe695d9e80bd31fa5b8ea266  variables-1-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-2-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-3-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-4-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-5-7.txt
68668da7fbb602eba00272eb60ed1c9b  variables-1-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-2-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-3-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-4-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-5-9.txt
6d91d159878c892dfd61782afd2879e5  variables-1-2.txt
6d91d159878c892dfd61782afd2879e5  variables-2-2.txt
6d91d159878c892dfd61782afd2879e5  variables-3-2.txt
6d91d159878c892dfd61782afd2879e5  variables-4-2.txt
6d91d159878c892dfd61782afd2879e5  variables-5-2.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-1-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-2-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-3-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-4-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-5-8.txt
92409d8058efa6d42f734d758f2149e8  variables-1-6.txt
92409d8058efa6d42f734d758f2149e8  variables-2-6.txt
92409d8058efa6d42f734d758f2149e8  variables-3-6.txt
92409d8058efa6d42f734d758f2149e8  variables-4-6.txt
92409d8058efa6d42f734d758f2149e8  variables-5-6.txt
9afda112d33708ba0fe68365d59eaf6c  variables-1-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-2-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-3-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-4-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-5-10.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-1-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-2-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-3-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-4-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-5-4.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-1-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-2-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-3-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-4-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-5-11.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-1-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-2-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-3-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-4-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-5-3.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-1-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-2-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-3-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-4-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-5-12.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-1-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-2-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-3-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-4-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-5-5.txt
e376ce0ed270136fcd035d30a6a87340  variables-1-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-2-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-3-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-4-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-5-13.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-1-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-2-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-3-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-4-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-5-1.txt
```

guilpier-code commented:
> 
blabla

guilpier-code commented:
About the **variables-\<year\>-\<week\>.txt** : 
these files only differ because of the last piece of information on each line.
This last piece is the hour in the year (1 to 3760).
So **variables-1-1.txt** and **variables-1-2.txt** are different.
But **variables-1-1.txt** and **variables-2-1.txt** are not.

So I recommend that we remove this last piece of information from each line of **variables-\<year\>-\<week\>.txt**, rename it **variables.txt**, and print this file only at the first week of the simulation, just like **interconnexion.txt** and **area.txt**

guilpier-code committed:
opt export structure : simplifying build of variables.txt + corrections after review

flomnes committed:
De-duplicate OPT_Export_add_variable by using `std::optional<int>` (#1042)

* De-duplicate OPT_Export_add_variable (std::optional<int>)

* PR remarks

guilpier-code committed:
Refactor : move current year and week from study to weekly optimization problem (#1032)

* Moving current year and week from study to weekly optimization problem

Why ?
- Because weekly problem is more the location of current year and week (the problem knows what year and week it's solving)
- It makes the code cleaner (reduces dependency on the study)
- It makes easier some changes to come (export MPS files when optimization is daily)

Some collateral cleaning was made

* move year and week from study to weekly opt problem : correction after sonar Q review

* move year and week from study to weekly opt problem : correction after human review

* Refactor : about optimization number (#1040)

* Refactor : about optimization number

- Information of optimization number is spread at 2 locations : it can be found inside ProblemeHebdo and inside ProblemeHebdo->ProblemeAResoudre
- This piece of information does not need to be nested in these structures
So it is declared and defined once, and locally, and then passed to lower functions when needed.

* Refactor : about optimization number : oops, correction of a unexpected remove

* Refactor : about optimization number : removing unnecessary trailing affectation

* [skip ci] Refactor about optimization number : rename the optimization number from french to english

flomnes committed:
Merge remote-tracking branch 'github/develop' into fix/refactor-export-structure

flomnes committed:
Fix build

flomnes committed:
Fix compilation warning

flomnes committed:
Restore area-1-1.txt, interco-1-1.txt, variables-Y-W.txt (#1046)

payetvin committed:
[MERGE] Merged develop and resolver conflicts in opt_gestion_des_bornes_cas_lineaire.cpp opt_gestion_second_membre_cas_lineaire.cpp

guilpier-code committed:
MPS files when a daily optimization is asked (#1041)

* Moving current year and week from study to weekly optimization problem

Why ?
- Because weekly problem is more the location of current year and week (the problem knows what year and week it's solving)
- It makes the code cleaner (reduces dependency on the study)
- It makes easier some changes to come (export MPS files when optimization is daily)

Some collateral cleaning was made

* move year and week from study to weekly opt problem : correction after sonar Q review

* move year and week from study to weekly opt problem : correction after human review

* Refactor : about optimization number

- Information of optimization number is spread at 2 locations : it can be found inside ProblemeHebdo and inside ProblemeHebdo->ProblemeAResoudre
- This piece of information does not need to be nested in these structures
So it is declared and defined once, and locally, and then passed to lower functions when needed.

* Refactor : about optimization number : oops, correction of a unexpected remove

* Refactor : about optimization number : removing unnecessary trailing affectation

* MPS when daily optimization : code simplifications

- unnecessary argument in the mpsWriterFactory's constructor
- we don't need to test optNumber in function that build the MPS and criterion file names

* MPS when daily optimization : code now produces 14 MPS files per week

* MPS when daily optimization : adding "override" key words

* MPS when daily optimization : in MPS files' name, change the order : day-week-year ==> year-day-week

* MPS when daily optimization : corrections after sonar Q (and human) review

* [skip ci] MPS when daily optimization : add comments on optimization period (as string)

* MPS when daily optimization : logging a MPS / criterion filename only if we actually print the associated file

* Fix & add tests related to MPS files (simplex-range = day, week)

* Fix MPS tests on Windows (#1045)

* Upload directory src/tests/mps in case of error

* Ignore CR;LF in Windows

* Revert spurious log

* Pass arg by const reference

* [skip ci] MPS when daily optimization : corrections after review

* [skip ci] MPS when daily optimization : corrections after review

* MPS when daily optimization : moving abstraction "optPeriodAsString" into a separate header

This to avoid a dependency to the filename.h's implementation code :
when changing filename.h's, implementation code, clients of "optPeriodAsString" do not recompile

* MPS when daily optimization : extracting optimization period as a string in .h/.cpp files

* Restore function getFilenameWithExtension

* MPS when daily optimization : renaming "opt period as string" into "opt period string generator"

* [skip ci] MPS when daily optimization : renaming a local variable

* [skip ci] MPS when daily optimization : some more renaming (classes and a local variable)

* [skip ci] Rename opt_period_as_string -> optPeriodStringGenerator

Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>
